crypto/pkcs7/pk7_attr.c

```diff
@@ -74,7 +74,10 @@ int PKCS7_simple_smimecap(STACK_OF(X509_ALGOR) *sk, int nid, int arg)
         alg->parameter->value.integer = nbit;
         alg->parameter->type = V_ASN1_INTEGER;
     }
-    sk_X509_ALGOR_push(sk, alg);
+    if (!sk_X509_ALGOR_push(sk, alg)) {
```

Comment:
Maybe not within the scope of this MR, but I note that mem handling in `if` block above looks like it could leak `alg` in the event of error.